.github/workflows/rust.yml

```diff
@@ -32,6 +32,26 @@ jobs:
     - name: Check style
       run: cargo fmt -- --check
 
+  check-omicron-deployment:
+    needs: skip_duplicate_jobs
+    if: ${{ needs.skip_duplicate_jobs.outputs.should_skip != 'true' }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-18.04, macos-11 ]
+        # See rust-toolchain for why we're using nightly here.
+        toolchain: [ nightly-2021-11-24 ]
```

Comment:
Same question elsewhere - isn't this inferred from `rust-toolchain.toml`? Why repeat the toolchain value?

(I'm nagging on this because re-stating the toolchain version in multiple spots seems like it can lead to them being out-of-sync -- I understand de-duping isn't always possible, but that's why I'm pushing in this direction)

Reply:
Yeah, single source of truth and all that, I hear you. This actually looks like a holdover from when we tested in a matrix with multiple toolchains. Since we're fixing one in the toolchain file, maybe we should remove this entirely. Thoughts?

Reply:
@davepacheco What do you think about removing the toolchain from the CI matrix, relying on the toolchain file alone?

Reply:
Yes.  I looked into this recently in some offline discussion re: #448.  I think the summary is:

- In April 2020, I added the GitHub workflow. I didn't know about rust-toolchain, I suspect, because I used the workflow config to control which toolchain the build would use. There was no rust-toolchain file in omicron at this point.
- In June 2021, we added the rust-toolchain file. At this point, I suspect we could have removed the config from the GitHub action.

Reply:
The consensus is to remove the references to the toolchain from the CI files, relying on the contents of `rust-toolchain.toml`. I've pushed a new commit with that change, and I'll verify that the right compiler (nightly-2021-11-24) is actually used. If not, I'll revert.

Reply:
Ok, I've confirmed that every job, including the Buildomat stuff, is using `2021-11-24` from the toolchain file. I think we're good here.